lib/omniauth/strategies/openid_connect.rb

```diff
@@ -196,9 +196,13 @@ def authorize_uri
       end
 
       def public_key
-        return config.jwks if options.discovery
-
-        key_or_secret || config.jwks
+        @public_key ||= if options.discovery
+                          config.jwks
+                        elsif key_or_secret
+                          key_or_secret
+                        elsif client_options.jwks_uri
+                          fetch_key
+                        end
```

Comment:
**Question**: should there be an `else` option here that returns `config.jwks`?

Previous logic would return `config.jwks if options.discovery` but also fell back to `config.jwks` if `key_or_secret` were `nil`. 

In addition to adding `fetch_key`, we are changing the fallback/default behavior here, which could have unintended / backward incompatible consequences. 

What do you think?

Reply:
@jessieay It's a good question, but `config` always attempts to call out to the OpenID discovery URL, which sort of defeats the purpose if discovery is disabled.

Reply:
ah ok that makes sense @stanhu . In other words: the former default return value was problematic and what we are fixing 💯